[{"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/788", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/788/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/788/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/788/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/788", "id": 731175475, "node_id": "MDU6SXNzdWU3MzExNzU0NzU=", "number": 788, "title": "W&B error due to Sliding Window", "user": {"login": "Yannik1337", "id": 50734087, "node_id": "MDQ6VXNlcjUwNzM0MDg3", "avatar_url": "https://avatars.githubusercontent.com/u/50734087?v=4", "gravatar_id": "", "url": "https://api.github.com/users/Yannik1337", "html_url": "https://github.com/Yannik1337", "followers_url": "https://api.github.com/users/Yannik1337/followers", "following_url": "https://api.github.com/users/Yannik1337/following{/other_user}", "gists_url": "https://api.github.com/users/Yannik1337/gists{/gist_id}", "starred_url": "https://api.github.com/users/Yannik1337/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/Yannik1337/subscriptions", "organizations_url": "https://api.github.com/users/Yannik1337/orgs", "repos_url": "https://api.github.com/users/Yannik1337/repos", "events_url": "https://api.github.com/users/Yannik1337/events{/privacy}", "received_events_url": "https://api.github.com/users/Yannik1337/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}, {"id": 1807729876, "node_id": "MDU6TGFiZWwxODA3NzI5ODc2", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/stale", "name": "stale", "color": "fcf8a9", "default": false, "description": "This issue has become stale"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2020-10-28T07:15:44Z", "updated_at": "2021-01-08T15:20:39Z", "closed_at": "2021-01-08T15:20:39Z", "author_association": "NONE", "active_lock_reason": null, "body": "**Describe the bug**\r\nRunning Bert on text input with sliding window enabled leads to an error in W&B generating a ROC curve. Not using sliding window works fine (but yields lower results)\r\n\r\n**Expected behavior**\r\nExpected no error. \r\n\r\n**Desktop \r\n- running on Linux, with python 3.7, and simple transformer 0.48.14\r\n\r\n**Additional context**\r\nTraceback (most recent call last): \r\n  File \".../scripts/simpletransformer.py\", line 70, in <module> \r\n    result, model_outputs, wrong_predictions = model.eval_model(test_df, acc=sklearn.metrics.accuracy_score, f1_macro=f1_func_macro, f1_micro=f1_func_micro) \r\n  File \"/opt/conda/lib/python3.7/site-packages/simpletransformers/classification/classification_model.py\", line 808, in eval_model \r\n    eval_df, output_dir, multi_label=multi_label, verbose=verbose, silent=silent, wandb_log=wandb_log, **kwargs \r\n  File \"/opt/conda/lib/python3.7/site-packages/simpletransformers/classification/classification_model.py\", line 974, in evaluate \r\n    wandb.log({\"roc\": wandb.plots.ROC(truth, model_outputs, labels_list)}) \r\n  File \"/opt/conda/lib/python3.7/site-packages/wandb/plots/roc.py\", line 73, in roc \r\n    return roc_table(fpr_dict, tpr_dict, classes, indices_to_plot) \r\n  File \"/opt/conda/lib/python3.7/site-packages/wandb/plots/roc.py\", line 52, in roc_table \r\n    fpr_dict[i], tpr_dict[i], _ = roc_curve(y_true, probas[:, i], \r\nIndexError: too many indices for array\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/788/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/788/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/696", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/696/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/696/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/696/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/696", "id": 694797896, "node_id": "MDU6SXNzdWU2OTQ3OTc4OTY=", "number": 696, "title": "After last update in ConvAIModel, getting AttributeError: 'ConvAIModel' object has no attribute 'model' while loading pre-trained", "user": {"login": "anindyasdas", "id": 50780535, "node_id": "MDQ6VXNlcjUwNzgwNTM1", "avatar_url": "https://avatars.githubusercontent.com/u/50780535?v=4", "gravatar_id": "", "url": "https://api.github.com/users/anindyasdas", "html_url": "https://github.com/anindyasdas", "followers_url": "https://api.github.com/users/anindyasdas/followers", "following_url": "https://api.github.com/users/anindyasdas/following{/other_user}", "gists_url": "https://api.github.com/users/anindyasdas/gists{/gist_id}", "starred_url": "https://api.github.com/users/anindyasdas/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/anindyasdas/subscriptions", "organizations_url": "https://api.github.com/users/anindyasdas/orgs", "repos_url": "https://api.github.com/users/anindyasdas/repos", "events_url": "https://api.github.com/users/anindyasdas/events{/privacy}", "received_events_url": "https://api.github.com/users/anindyasdas/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2020-09-07T07:24:11Z", "updated_at": "2020-09-13T09:40:41Z", "closed_at": "2020-09-13T09:40:41Z", "author_association": "NONE", "active_lock_reason": null, "body": "**Describe the bug**\r\nWhile loading the pre-trained model  getting error \"AttributeError: 'ConvAIModel' object has no attribute 'model'\"\r\n\r\nThis error appears appears after the recent changes made.\r\n\r\n**To Reproduce**\r\n\r\n`model = ConvAIModel(\"gpt\",model_path, use_cuda=True, args=train_args)`\r\n\r\n**Expected behavior**\r\nThis error was not there before the update. It seems like it refers self.model before assigning self.model \r\n\r\n**Screenshots**\r\n`AttributeError                            Traceback (most recent call last)\r\n<ipython-input-53-1daa579c40bc> in <module>()\r\n----> 1 model = ConvAIModel(\"gpt\",model_path, use_cuda=True, args=train_args)\r\n\r\n/usr/local/lib/python3.6/dist-packages/simpletransformers/conv_ai/conv_ai_model.py in __init__(self, model_type, model_name, args, use_cuda, cuda_device, **kwargs)\r\n    117         self.__dict__.update(kwargs)\r\n    118         self.tokenizer = tokenizer_class.from_pretrained(model_name, **kwargs)\r\n--> **119         self.add_special_tokens_(self.model, self.tokenizer)**\r\n    120 \r\n    121         if not self.args.quantized_model:\r\n\r\nAttributeError: 'ConvAIModel' object has no attribute 'model`\r\n\r\n@ThilinaRajapakse ", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/696/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/696/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/665", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/665/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/665/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/665/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/665", "id": 683418094, "node_id": "MDU6SXNzdWU2ODM0MTgwOTQ=", "number": 665, "title": "when set pos_weights for multilabel-classification,it can not train on multi-gpus", "user": {"login": "MrRace", "id": 10300313, "node_id": "MDQ6VXNlcjEwMzAwMzEz", "avatar_url": "https://avatars.githubusercontent.com/u/10300313?v=4", "gravatar_id": "", "url": "https://api.github.com/users/MrRace", "html_url": "https://github.com/MrRace", "followers_url": "https://api.github.com/users/MrRace/followers", "following_url": "https://api.github.com/users/MrRace/following{/other_user}", "gists_url": "https://api.github.com/users/MrRace/gists{/gist_id}", "starred_url": "https://api.github.com/users/MrRace/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/MrRace/subscriptions", "organizations_url": "https://api.github.com/users/MrRace/orgs", "repos_url": "https://api.github.com/users/MrRace/repos", "events_url": "https://api.github.com/users/MrRace/events{/privacy}", "received_events_url": "https://api.github.com/users/MrRace/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}, {"id": 1807729876, "node_id": "MDU6TGFiZWwxODA3NzI5ODc2", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/stale", "name": "stale", "color": "fcf8a9", "default": false, "description": "This issue has become stale"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 4, "created_at": "2020-08-21T09:12:45Z", "updated_at": "2020-12-31T01:24:57Z", "closed_at": "2020-12-31T01:24:57Z", "author_association": "NONE", "active_lock_reason": null, "body": "**Describe the bug**\r\nFor `multilabel_classification.py` the `pos_weights` does not support training on multi-gpus.\r\n\r\n**To Reproduce**\r\nUse the example from `simpletransformers\\examples\\text_classification\\multilabel_classification.py`. When assign my weights of classes to  `pos_weight` . I set  `n_gpu=3`. When train with multi-gpus it comes the error\uff1a\r\n```\r\nExpected all tensors to be on the same device, but found at least two devices, cuda:0 and cuda:1\r\n```\r\n\r\nWhen do not pass my own pos_weights to parameter `pos_weight`, it can train on multi-gpus normally.\r\n\r\nWhen pass my own pos_weights to parameter `pos_weight` and set it  `n_gpu=1` it can train  normally, but just use single GPU.\r\n\r\n**Screenshots**\r\n![image](https://user-images.githubusercontent.com/10300313/90873390-b1911700-e3d0-11ea-8c9e-ad5f9c0ea5da.png)\r\n\r\n**environment (please complete the following information):**\r\n - Linux\r\n- Python 3.7\r\n\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/665/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/665/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/657", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/657/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/657/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/657/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/657", "id": 680635847, "node_id": "MDU6SXNzdWU2ODA2MzU4NDc=", "number": 657, "title": "Trained model is not saved to `output_dir` parameter directory.", "user": {"login": "jnishi", "id": 836541, "node_id": "MDQ6VXNlcjgzNjU0MQ==", "avatar_url": "https://avatars.githubusercontent.com/u/836541?v=4", "gravatar_id": "", "url": "https://api.github.com/users/jnishi", "html_url": "https://github.com/jnishi", "followers_url": "https://api.github.com/users/jnishi/followers", "following_url": "https://api.github.com/users/jnishi/following{/other_user}", "gists_url": "https://api.github.com/users/jnishi/gists{/gist_id}", "starred_url": "https://api.github.com/users/jnishi/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/jnishi/subscriptions", "organizations_url": "https://api.github.com/users/jnishi/orgs", "repos_url": "https://api.github.com/users/jnishi/repos", "events_url": "https://api.github.com/users/jnishi/events{/privacy}", "received_events_url": "https://api.github.com/users/jnishi/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}, {"id": 1807729876, "node_id": "MDU6TGFiZWwxODA3NzI5ODc2", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/stale", "name": "stale", "color": "fcf8a9", "default": false, "description": "This issue has become stale"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2020-08-18T02:15:02Z", "updated_at": "2020-10-24T23:16:53Z", "closed_at": "2020-10-24T23:16:53Z", "author_association": "NONE", "active_lock_reason": null, "body": "**Describe the bug**\r\nThe trained model of ClassificationModel is saved to the directory  where the `output_dir` parameter of `ClassificationModel.train_model` function. However, a part of trained model (e.g. config.json, pytorch_model.bin, etc.) is saved to the directory which is specified by `args.output_dir` even if  `output_dir` parameter of  `train_model` is specified.\r\n\r\n**Expected behavior**\r\nAll of trained model is saved to the directory which is specified by `output_dir` parameter.\r\n\r\n**Desktop (please complete the following information):**\r\n - OS:  Ubuntu 16.04.6 LTS\r\n\r\n**Additional context**\r\nThis may be because `_save_model` which is called in `train_model` does not use `output_dir` variable in `train_model`.\r\nhttps://github.com/ThilinaRajapakse/simpletransformers/blob/v0.47.2/simpletransformers/classification/classification_model.py#L313", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/657/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/657/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/588", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/588/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/588/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/588/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/588", "id": 664080844, "node_id": "MDU6SXNzdWU2NjQwODA4NDQ=", "number": 588, "title": "Distributed training issue in language_modeling_model.py", "user": {"login": "abhik1505040", "id": 49608995, "node_id": "MDQ6VXNlcjQ5NjA4OTk1", "avatar_url": "https://avatars.githubusercontent.com/u/49608995?v=4", "gravatar_id": "", "url": "https://api.github.com/users/abhik1505040", "html_url": "https://github.com/abhik1505040", "followers_url": "https://api.github.com/users/abhik1505040/followers", "following_url": "https://api.github.com/users/abhik1505040/following{/other_user}", "gists_url": "https://api.github.com/users/abhik1505040/gists{/gist_id}", "starred_url": "https://api.github.com/users/abhik1505040/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/abhik1505040/subscriptions", "organizations_url": "https://api.github.com/users/abhik1505040/orgs", "repos_url": "https://api.github.com/users/abhik1505040/repos", "events_url": "https://api.github.com/users/abhik1505040/events{/privacy}", "received_events_url": "https://api.github.com/users/abhik1505040/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}, {"id": 1807729876, "node_id": "MDU6TGFiZWwxODA3NzI5ODc2", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/stale", "name": "stale", "color": "fcf8a9", "default": false, "description": "This issue has become stale"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2020-07-22T22:11:04Z", "updated_at": "2020-09-30T03:10:43Z", "closed_at": "2020-09-30T03:10:43Z", "author_association": "NONE", "active_lock_reason": null, "body": "[https://github.com/ThilinaRajapakse/simpletransformers/blob/master/simpletransformers/language_modeling/language_modeling_model.py#L149-L166](https://github.com/ThilinaRajapakse/simpletransformers/blob/master/simpletransformers/language_modeling/language_modeling_model.py#L149-L166)\r\n\r\nShouln't this block be inside the `train_model` function? In a distributed setting, the tokenizer would be trained unnecessarily for multiple times.", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/588/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/588/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/585", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/585/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/585/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/585/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/585", "id": 663533770, "node_id": "MDU6SXNzdWU2NjM1MzM3NzA=", "number": 585, "title": "Missing labels on NER eval_model with classification_report set to True", "user": {"login": "yoeldk", "id": 12507451, "node_id": "MDQ6VXNlcjEyNTA3NDUx", "avatar_url": "https://avatars.githubusercontent.com/u/12507451?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yoeldk", "html_url": "https://github.com/yoeldk", "followers_url": "https://api.github.com/users/yoeldk/followers", "following_url": "https://api.github.com/users/yoeldk/following{/other_user}", "gists_url": "https://api.github.com/users/yoeldk/gists{/gist_id}", "starred_url": "https://api.github.com/users/yoeldk/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yoeldk/subscriptions", "organizations_url": "https://api.github.com/users/yoeldk/orgs", "repos_url": "https://api.github.com/users/yoeldk/repos", "events_url": "https://api.github.com/users/yoeldk/events{/privacy}", "received_events_url": "https://api.github.com/users/yoeldk/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}, {"id": 1807729876, "node_id": "MDU6TGFiZWwxODA3NzI5ODc2", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/stale", "name": "stale", "color": "fcf8a9", "default": false, "description": "This issue has become stale"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 4, "created_at": "2020-07-22T07:15:44Z", "updated_at": "2020-12-03T05:53:53Z", "closed_at": "2020-12-03T05:53:53Z", "author_association": "NONE", "active_lock_reason": null, "body": "**Describe the bug**\r\nIt seems like the labels shown in the report are taken from the (true) labels in the data to be evaluated on instead of the list of all labels the model was configured with. \r\nIf the evaluation is done on a (smaller) dataset that does not contain all possible labels the model was originally trained on (let's call the set of missing labels {Y}), then false positives of labels that belong to {Y} do not appear in the statistics. \r\n\r\n**To Reproduce**\r\nEvaluate NER model with classification_report on a data set that has predicted labels which do not appear in the input data as true positives.\r\n \r\n**Expected behavior**\r\nThe expected behavior would be to produce a classification_report for each label in the **original list of labels the model was configured with** instead of the true labels in the data to be evaluated on.", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/585/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/585/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/558", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/558/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/558/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/558/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/558", "id": 653389609, "node_id": "MDU6SXNzdWU2NTMzODk2MDk=", "number": 558, "title": "RuntimeError: expected device cuda:0 but got device cuda:1", "user": {"login": "elulue", "id": 16643210, "node_id": "MDQ6VXNlcjE2NjQzMjEw", "avatar_url": "https://avatars.githubusercontent.com/u/16643210?v=4", "gravatar_id": "", "url": "https://api.github.com/users/elulue", "html_url": "https://github.com/elulue", "followers_url": "https://api.github.com/users/elulue/followers", "following_url": "https://api.github.com/users/elulue/following{/other_user}", "gists_url": "https://api.github.com/users/elulue/gists{/gist_id}", "starred_url": "https://api.github.com/users/elulue/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/elulue/subscriptions", "organizations_url": "https://api.github.com/users/elulue/orgs", "repos_url": "https://api.github.com/users/elulue/repos", "events_url": "https://api.github.com/users/elulue/events{/privacy}", "received_events_url": "https://api.github.com/users/elulue/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2020-07-08T15:26:17Z", "updated_at": "2020-08-03T17:02:46Z", "closed_at": "2020-08-03T17:02:46Z", "author_association": "NONE", "active_lock_reason": null, "body": "\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/558/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/558/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/531", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/531/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/531/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/531/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/531", "id": 649358567, "node_id": "MDU6SXNzdWU2NDkzNTg1Njc=", "number": 531, "title": "Encoder Decoder seq2seq Tokenization and 'lm_labels' Error", "user": {"login": "ahmetbagci8", "id": 36896422, "node_id": "MDQ6VXNlcjM2ODk2NDIy", "avatar_url": "https://avatars.githubusercontent.com/u/36896422?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ahmetbagci8", "html_url": "https://github.com/ahmetbagci8", "followers_url": "https://api.github.com/users/ahmetbagci8/followers", "following_url": "https://api.github.com/users/ahmetbagci8/following{/other_user}", "gists_url": "https://api.github.com/users/ahmetbagci8/gists{/gist_id}", "starred_url": "https://api.github.com/users/ahmetbagci8/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ahmetbagci8/subscriptions", "organizations_url": "https://api.github.com/users/ahmetbagci8/orgs", "repos_url": "https://api.github.com/users/ahmetbagci8/repos", "events_url": "https://api.github.com/users/ahmetbagci8/events{/privacy}", "received_events_url": "https://api.github.com/users/ahmetbagci8/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 6, "created_at": "2020-07-01T22:16:42Z", "updated_at": "2020-12-01T04:16:48Z", "closed_at": "2020-07-10T12:21:39Z", "author_association": "NONE", "active_lock_reason": null, "body": "**Describe the bug**\r\nHello,\r\nWhen I train Encoder Decoder seq2seq, I get an error. Error message is \r\n\r\nWARNING:transformers.tokenization_utils_base:Truncation was not explicitely activated but `max_length` is provided a specific value, please use `truncation=True` to explicitely truncate examples to max length. Defaulting to 'only_first' truncation strategy. If you encode pairs of sequences (GLUE-style) with the tokenizer you may want to check this is the right behavior.\r\n\r\nI suppress this warning to change transformers_logger.setLevel(logging.WARNING) line to transformers_logger.setLevel(logging.ERROR) . After that, I get this error \r\n\r\nTypeError: forward() got an unexpected keyword argument 'lm_labels'\r\n\r\nI had no problem 2 days ago. Can you help me?\r\n\r\nThanks.\r\n\r\n**To Reproduce**\r\nYou can try Generic Encoder-Decoder minimal start. I get same errors.\r\n\r\n**Desktop (please complete the following information):**\r\n - Google Colab\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/531/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/531/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/529", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/529/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/529/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/529/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/529", "id": 649113023, "node_id": "MDU6SXNzdWU2NDkxMTMwMjM=", "number": 529, "title": "'tuple' object does not support item assignment | Multimodal eval", "user": {"login": "AshishGogo", "id": 30543644, "node_id": "MDQ6VXNlcjMwNTQzNjQ0", "avatar_url": "https://avatars.githubusercontent.com/u/30543644?v=4", "gravatar_id": "", "url": "https://api.github.com/users/AshishGogo", "html_url": "https://github.com/AshishGogo", "followers_url": "https://api.github.com/users/AshishGogo/followers", "following_url": "https://api.github.com/users/AshishGogo/following{/other_user}", "gists_url": "https://api.github.com/users/AshishGogo/gists{/gist_id}", "starred_url": "https://api.github.com/users/AshishGogo/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/AshishGogo/subscriptions", "organizations_url": "https://api.github.com/users/AshishGogo/orgs", "repos_url": "https://api.github.com/users/AshishGogo/repos", "events_url": "https://api.github.com/users/AshishGogo/events{/privacy}", "received_events_url": "https://api.github.com/users/AshishGogo/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 6, "created_at": "2020-07-01T16:26:07Z", "updated_at": "2020-07-02T20:26:31Z", "closed_at": "2020-07-02T20:26:31Z", "author_association": "NONE", "active_lock_reason": null, "body": "**Describe the bug**\r\nGetting error **'tuple' object does not support item assignment** while running eval_model on MultiModalClassificationModel(\"bert\", \"bert-base-uncased\",label_list=[0,1])\r\n\r\n**To Reproduce**\r\nrun eval_model on any text+images multimodal problem.\r\n\r\n**Expected behavior**\r\nshould have provided predictions\r\n\r\n**Screenshots**\r\nIf applicable, add screenshots to help explain your problem.\r\n\r\n**Desktop (please complete the following information):**\r\n - OS\r\n\r\n**Additional context**\r\nAdd any other context about the problem here.\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/529/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/529/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/501", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/501/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/501/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/501/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/501", "id": 643361087, "node_id": "MDU6SXNzdWU2NDMzNjEwODc=", "number": 501, "title": "MultiModalClassificationModel eval_data issue", "user": {"login": "caprone", "id": 15055331, "node_id": "MDQ6VXNlcjE1MDU1MzMx", "avatar_url": "https://avatars.githubusercontent.com/u/15055331?v=4", "gravatar_id": "", "url": "https://api.github.com/users/caprone", "html_url": "https://github.com/caprone", "followers_url": "https://api.github.com/users/caprone/followers", "following_url": "https://api.github.com/users/caprone/following{/other_user}", "gists_url": "https://api.github.com/users/caprone/gists{/gist_id}", "starred_url": "https://api.github.com/users/caprone/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/caprone/subscriptions", "organizations_url": "https://api.github.com/users/caprone/orgs", "repos_url": "https://api.github.com/users/caprone/repos", "events_url": "https://api.github.com/users/caprone/events{/privacy}", "received_events_url": "https://api.github.com/users/caprone/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 7, "created_at": "2020-06-22T21:06:49Z", "updated_at": "2020-06-30T21:00:51Z", "closed_at": "2020-06-30T21:00:51Z", "author_association": "NONE", "active_lock_reason": null, "body": "HI\r\nwhen I try to evaluate model during training( df_train is an dataframe):\r\nmodel = MultiModalClassificationModel(model_type, model_name, num_labels=2, args=model_args)\r\n\r\nmodel.train_model(df_train[start0:end0, :], image_path=image_path, eval_data=df_train[start1:end1, :], **eval_metrics)\r\n\r\nit raises :\r\n\"\"\"ValueError: data is not a str and image_path is not given. image_path must be specified when input is a DF\"\"\"\r\n\r\nbut I have specified correctly image_path, and dataset is the same with different slices..; if I turn off 'evaluate_during_training', then everything go fine;\r\n\r\nsorry, but I am unable to find the problem...\r\n\r\nthanks\r\n\r\n\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/501/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/501/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/496", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/496/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/496/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/496/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/496", "id": 642584313, "node_id": "MDU6SXNzdWU2NDI1ODQzMTM=", "number": 496, "title": "For ELECTRA, masking 15% tokens should replace with [MASK], not include random word", "user": {"login": "dev-sngwn", "id": 33826207, "node_id": "MDQ6VXNlcjMzODI2MjA3", "avatar_url": "https://avatars.githubusercontent.com/u/33826207?v=4", "gravatar_id": "", "url": "https://api.github.com/users/dev-sngwn", "html_url": "https://github.com/dev-sngwn", "followers_url": "https://api.github.com/users/dev-sngwn/followers", "following_url": "https://api.github.com/users/dev-sngwn/following{/other_user}", "gists_url": "https://api.github.com/users/dev-sngwn/gists{/gist_id}", "starred_url": "https://api.github.com/users/dev-sngwn/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/dev-sngwn/subscriptions", "organizations_url": "https://api.github.com/users/dev-sngwn/orgs", "repos_url": "https://api.github.com/users/dev-sngwn/repos", "events_url": "https://api.github.com/users/dev-sngwn/events{/privacy}", "received_events_url": "https://api.github.com/users/dev-sngwn/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2020-06-21T15:26:34Z", "updated_at": "2020-07-07T22:59:08Z", "closed_at": "2020-07-07T22:59:08Z", "author_association": "NONE", "active_lock_reason": null, "body": "\r\nhttps://github.com/ThilinaRajapakse/simpletransformers/blob/cf66100fafedc7a6277288d4eb1cf2aed6b3e676/simpletransformers/language_modeling/language_modeling_utils.py#L243\r\n\r\n\r\nIn [ELECTRA (Clark et al)](https://openreview.net/pdf?id=r1xMH1BtvB), they mentioned about this issue in **3.5 EFFICIENCY ANALYSIS**\r\n\r\n> masked tokens are replaced with a random token 10% of the time and are kept the same 10% of the time. However, our results suggest these simple heuristics are insufficient to fully solve the issue.\r\n\r\nSo, their [official implementation (google-research/electra)](https://github.com/google-research/electra/blob/81f7e5fc98b0ad8bfd20b641aa8bc9e6ac00c8eb/pretrain/pretrain_helpers.py#L188) does not including replacing random word process. Please check this out and let me know if I wrong.\r\n\r\n_Thanks & best regards,_\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/496/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/496/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/484", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/484/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/484/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/484/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/484", "id": 639872737, "node_id": "MDU6SXNzdWU2Mzk4NzI3Mzc=", "number": 484, "title": "ClassificationModel with output_hidden_states=True results in IndexError", "user": {"login": "N8Brooks", "id": 20604441, "node_id": "MDQ6VXNlcjIwNjA0NDQx", "avatar_url": "https://avatars.githubusercontent.com/u/20604441?v=4", "gravatar_id": "", "url": "https://api.github.com/users/N8Brooks", "html_url": "https://github.com/N8Brooks", "followers_url": "https://api.github.com/users/N8Brooks/followers", "following_url": "https://api.github.com/users/N8Brooks/following{/other_user}", "gists_url": "https://api.github.com/users/N8Brooks/gists{/gist_id}", "starred_url": "https://api.github.com/users/N8Brooks/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/N8Brooks/subscriptions", "organizations_url": "https://api.github.com/users/N8Brooks/orgs", "repos_url": "https://api.github.com/users/N8Brooks/repos", "events_url": "https://api.github.com/users/N8Brooks/events{/privacy}", "received_events_url": "https://api.github.com/users/N8Brooks/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}, {"id": 1807729876, "node_id": "MDU6TGFiZWwxODA3NzI5ODc2", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/stale", "name": "stale", "color": "fcf8a9", "default": false, "description": "This issue has become stale"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2020-06-16T18:06:29Z", "updated_at": "2020-08-29T18:39:24Z", "closed_at": "2020-08-29T18:39:24Z", "author_association": "NONE", "active_lock_reason": null, "body": "**Problem**\r\nI'm using the classification model with the config={'output_hidden_states':True}. When I predict for a set it creates an error with 'tuple index out of range.\r\n\r\n**Here is the code:**\r\n`from simpletransformers.classification import ClassificationModel\r\n\r\nargs = dict(\r\n            overwrite_output_dir=True, \r\n            max_seq_length=512,\r\n            train_batch_size=8,\r\n            test_batch_size=8,\r\n            gradient_accumulation_steps=2,\r\n            num_train_epochs=1,\r\n            sliding_window=True,\r\n            use_early_stopping=False,\r\n            config=dict(output_hidden_states=True)\r\n           )\r\n\r\nmodel = ClassificationModel('electra', 'google/electra-base-discriminator', num_labels=5, args=args)\r\n\r\nmodel.train_model(train, show_running_loss=True, eval_df=val)\r\n\r\ntest['pred'], *_ = model.predict(test['text'])`\r\n\r\n**The error:**\r\nTraceback (most recent call last):\r\n\r\n  File \"<ipython-input-25-6a26f3dc4ca6>\", line 1, in <module>\r\n    model.predict(test['text'])\r\n\r\n  File \"/home/nathan/anaconda3/envs/py38/lib/python3.8/site-packages/simpletransformers/classification/classification_model.py\", line 976, in predict\r\n    embedding_outputs, layer_hidden_states = outputs[2][0], outputs[2][1:]\r\n\r\nIndexError: tuple index out of range\r\n\r\n**Additional context**\r\nI'm trying to get the document embedding from the training data. I was planning on using the average of the second to last hidden layer. If you have any other ideas to achieve the same thing let me know :P\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/484/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/484/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/444", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/444/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/444/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/444/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/444", "id": 629002751, "node_id": "MDU6SXNzdWU2MjkwMDI3NTE=", "number": 444, "title": "Electra finetuning issues?", "user": {"login": "ddofer", "id": 5120387, "node_id": "MDQ6VXNlcjUxMjAzODc=", "avatar_url": "https://avatars.githubusercontent.com/u/5120387?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ddofer", "html_url": "https://github.com/ddofer", "followers_url": "https://api.github.com/users/ddofer/followers", "following_url": "https://api.github.com/users/ddofer/following{/other_user}", "gists_url": "https://api.github.com/users/ddofer/gists{/gist_id}", "starred_url": "https://api.github.com/users/ddofer/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ddofer/subscriptions", "organizations_url": "https://api.github.com/users/ddofer/orgs", "repos_url": "https://api.github.com/users/ddofer/repos", "events_url": "https://api.github.com/users/ddofer/events{/privacy}", "received_events_url": "https://api.github.com/users/ddofer/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}, {"id": 1807729876, "node_id": "MDU6TGFiZWwxODA3NzI5ODc2", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/stale", "name": "stale", "color": "fcf8a9", "default": false, "description": "This issue has become stale"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 7, "created_at": "2020-06-02T08:22:08Z", "updated_at": "2020-12-13T22:05:14Z", "closed_at": "2020-12-13T22:05:14Z", "author_association": "NONE", "active_lock_reason": null, "body": "**Describe the bug**\r\nI've trained a number of Electra language models (with 1-2 epochs) using Electra/LM pretraining, then saving the discrimniator. I've done this with different vocab sizes [80, 500, 15000] , and My text data is long sentences without any spaces). \r\n\r\nWhen I fine-tune the electra discriminator model on a new classification or multiclass task, performance is abysmal. It barely learns and just outputs the majority class. (I've tried training a bag of chars/n-grams, and a keras LSTM model on the tasks, and they performed very well). \r\n\r\nMy guess is that there is a bug with how the data is being processed in the fine tuning. \r\n**To Reproduce**\r\nDataset is private but shared via email. Long sentences without spaces, all upper cased. \r\nTrain electra language model, then try to fine tune discriminator on a new classification task. \r\n\r\n**Expected behavior**\r\nBetter than random results? \r\n\r\n**Desktop (please complete the following information):**\r\n - Windows 10. \r\nLatest transformers package (0.31 as of this time)\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/444/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/444/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/439", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/439/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/439/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/439/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/439", "id": 628340292, "node_id": "MDU6SXNzdWU2MjgzNDAyOTI=", "number": 439, "title": "Difference implementation of LineByLineTextDataset in Transformers and SimpleTransformers", "user": {"login": "cahya-wirawan", "id": 7669893, "node_id": "MDQ6VXNlcjc2Njk4OTM=", "avatar_url": "https://avatars.githubusercontent.com/u/7669893?v=4", "gravatar_id": "", "url": "https://api.github.com/users/cahya-wirawan", "html_url": "https://github.com/cahya-wirawan", "followers_url": "https://api.github.com/users/cahya-wirawan/followers", "following_url": "https://api.github.com/users/cahya-wirawan/following{/other_user}", "gists_url": "https://api.github.com/users/cahya-wirawan/gists{/gist_id}", "starred_url": "https://api.github.com/users/cahya-wirawan/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/cahya-wirawan/subscriptions", "organizations_url": "https://api.github.com/users/cahya-wirawan/orgs", "repos_url": "https://api.github.com/users/cahya-wirawan/repos", "events_url": "https://api.github.com/users/cahya-wirawan/events{/privacy}", "received_events_url": "https://api.github.com/users/cahya-wirawan/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 8, "created_at": "2020-06-01T10:47:58Z", "updated_at": "2020-07-01T07:30:24Z", "closed_at": "2020-07-01T07:30:24Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "Hi, thanks for the great wrapper for Transformers. Btw, I compared the implementation of LineByLineTextDataset in Transformers and SimpleTransformers, it seems they are difference. in SimpleTransformers you used tokenizer = ByteLevelBPETokenizer(),  and tokenizer._tokenizer.post_processor = BertProcessing(), but the Transformers use either the default tokenizer (PreTrainedTokenizer) or the tokenizer set in LineByLineTextDataset. If I don't misunderstood, your LineByLineTextDatasets implementation will only be working for Bert, is it correct?", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/439/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/439/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/386", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/386/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/386/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/386/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/386", "id": 621623750, "node_id": "MDU6SXNzdWU2MjE2MjM3NTA=", "number": 386, "title": "The results generated by SEQ2SEQ are meaningless.", "user": {"login": "ScottishFold007", "id": 36957508, "node_id": "MDQ6VXNlcjM2OTU3NTA4", "avatar_url": "https://avatars.githubusercontent.com/u/36957508?v=4", "gravatar_id": "", "url": "https://api.github.com/users/ScottishFold007", "html_url": "https://github.com/ScottishFold007", "followers_url": "https://api.github.com/users/ScottishFold007/followers", "following_url": "https://api.github.com/users/ScottishFold007/following{/other_user}", "gists_url": "https://api.github.com/users/ScottishFold007/gists{/gist_id}", "starred_url": "https://api.github.com/users/ScottishFold007/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/ScottishFold007/subscriptions", "organizations_url": "https://api.github.com/users/ScottishFold007/orgs", "repos_url": "https://api.github.com/users/ScottishFold007/repos", "events_url": "https://api.github.com/users/ScottishFold007/events{/privacy}", "received_events_url": "https://api.github.com/users/ScottishFold007/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 7, "created_at": "2020-05-20T09:50:24Z", "updated_at": "2020-07-01T07:30:50Z", "closed_at": "2020-07-01T07:30:50Z", "author_association": "NONE", "active_lock_reason": null, "body": "I tried to use the SEQ2SEQ model, but the results generated were meaningless:\r\n\r\n![image](https://user-images.githubusercontent.com/36957508/82431976-f6e83480-9ac1-11ea-8097-6075679e484e.png)\r\n![image](https://user-images.githubusercontent.com/36957508/82432102-2434e280-9ac2-11ea-9256-d132c04f1782.png)\r\n![image](https://user-images.githubusercontent.com/36957508/82432179-3dd62a00-9ac2-11ea-8cbc-1cd96625b0d5.png)\r\nwhy?", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/386/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/386/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/339", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/339/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/339/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/339/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/339", "id": 610316669, "node_id": "MDU6SXNzdWU2MTAzMTY2Njk=", "number": 339, "title": "\"early_stopping_metric\" doesn't seem to work", "user": {"login": "Lysimachos", "id": 18085954, "node_id": "MDQ6VXNlcjE4MDg1OTU0", "avatar_url": "https://avatars.githubusercontent.com/u/18085954?v=4", "gravatar_id": "", "url": "https://api.github.com/users/Lysimachos", "html_url": "https://github.com/Lysimachos", "followers_url": "https://api.github.com/users/Lysimachos/followers", "following_url": "https://api.github.com/users/Lysimachos/following{/other_user}", "gists_url": "https://api.github.com/users/Lysimachos/gists{/gist_id}", "starred_url": "https://api.github.com/users/Lysimachos/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/Lysimachos/subscriptions", "organizations_url": "https://api.github.com/users/Lysimachos/orgs", "repos_url": "https://api.github.com/users/Lysimachos/repos", "events_url": "https://api.github.com/users/Lysimachos/events{/privacy}", "received_events_url": "https://api.github.com/users/Lysimachos/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 12, "created_at": "2020-04-30T19:06:44Z", "updated_at": "2021-03-04T07:54:00Z", "closed_at": "2020-07-01T07:31:25Z", "author_association": "NONE", "active_lock_reason": null, "body": "I am trying multi_label and multi_class classification with early stopping.\r\n\r\nI have insert the sklearn.metrics.f1(average = \"micro) function into **kwargs with using this code:\r\n\r\n               ` kwargs = {\"f1_micro\": partial(sklearn.metrics.f1_score, average = \"micro\")}`\r\n\r\nThe f1_micro score is being calculated on the eval_df and it gets printed while training. But the argument :\r\n\r\n\"early_stopping_metric\" = \"f1_micro\" doesn't work. It seems that it always calculates the loss function on eval_df for the early stopping functionality.\r\nIt is my bad or there is indeed a bug?", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/339/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/339/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/337", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/337/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/337/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/337/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/337", "id": 610082592, "node_id": "MDU6SXNzdWU2MTAwODI1OTI=", "number": 337, "title": "Language model fine-tuning of dbmdz/bert-base-italian-xxl-cased'", "user": {"login": "freeIsa", "id": 61782307, "node_id": "MDQ6VXNlcjYxNzgyMzA3", "avatar_url": "https://avatars.githubusercontent.com/u/61782307?v=4", "gravatar_id": "", "url": "https://api.github.com/users/freeIsa", "html_url": "https://github.com/freeIsa", "followers_url": "https://api.github.com/users/freeIsa/followers", "following_url": "https://api.github.com/users/freeIsa/following{/other_user}", "gists_url": "https://api.github.com/users/freeIsa/gists{/gist_id}", "starred_url": "https://api.github.com/users/freeIsa/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/freeIsa/subscriptions", "organizations_url": "https://api.github.com/users/freeIsa/orgs", "repos_url": "https://api.github.com/users/freeIsa/repos", "events_url": "https://api.github.com/users/freeIsa/events{/privacy}", "received_events_url": "https://api.github.com/users/freeIsa/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 3, "created_at": "2020-04-30T14:40:41Z", "updated_at": "2020-07-01T07:32:09Z", "closed_at": "2020-07-01T07:32:09Z", "author_association": "NONE", "active_lock_reason": null, "body": "Hello, I am trying to use simpletransformers to fine-tune the Italian version of BERT ('dbmdz/bert-base-italian-xxl-cased') with my own dataset. As an exercise, I also would like to do the same directly with the transformers package. I have set the same seed for both projects, and tried to ensure that the language model is always trained with the same training_args, so I would expect to get the same results when running the two versions of my code. However, results are slightly different. Can you please suggest other sources of discrepancies I may have to look into?\r\n\r\nMoreover, I intended to use the \"line_by_line\" dataset_type, but I cannot get this to work under simpletransformers (whereas it does work when using transformers directly) with the above mentioned Bert model, as I get the following error message:\r\n\r\n```\r\n100         tokenizer = ByteLevelBPETokenizer(\r\n--> 101             f\"{args['tokenizer_name']}/vocab.json\", f\"{args['tokenizer_name']}/merges.txt\",\r\n    102         )\r\n[...]\r\nException: Error while initializing BPE: No such file or directory (os error 2)\r\n```\r\n\r\nThank you in advance for any help you can give me with these issues!", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/337/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/337/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/287", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/287/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/287/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/287/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/287", "id": 589403914, "node_id": "MDU6SXNzdWU1ODk0MDM5MTQ=", "number": 287, "title": "wandb RuntimeError", "user": {"login": "djstrong", "id": 1849959, "node_id": "MDQ6VXNlcjE4NDk5NTk=", "avatar_url": "https://avatars.githubusercontent.com/u/1849959?v=4", "gravatar_id": "", "url": "https://api.github.com/users/djstrong", "html_url": "https://github.com/djstrong", "followers_url": "https://api.github.com/users/djstrong/followers", "following_url": "https://api.github.com/users/djstrong/following{/other_user}", "gists_url": "https://api.github.com/users/djstrong/gists{/gist_id}", "starred_url": "https://api.github.com/users/djstrong/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/djstrong/subscriptions", "organizations_url": "https://api.github.com/users/djstrong/orgs", "repos_url": "https://api.github.com/users/djstrong/repos", "events_url": "https://api.github.com/users/djstrong/events{/privacy}", "received_events_url": "https://api.github.com/users/djstrong/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}, {"id": 1807729876, "node_id": "MDU6TGFiZWwxODA3NzI5ODc2", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/stale", "name": "stale", "color": "fcf8a9", "default": false, "description": "This issue has become stale"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 6, "created_at": "2020-03-27T20:45:22Z", "updated_at": "2020-07-13T22:04:28Z", "closed_at": "2020-06-03T10:09:31Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "When training text classification models using xlnet-large-cased, albert-base-v2, xlnet-base-cased and wandb enabled:\r\n```\r\nFile \"train.py\", line 101, in <module>\r\n    rc=sklearn.metrics.recall_score)\r\n  File \"venv/lib/python3.7/site-packages/simpletransformers/classification/classification_model.py\", line 267, in \r\ntrain_model\r\n    **kwargs,\r\n  File \"venv/lib/python3.7/site-packages/simpletransformers/classification/classification_model.py\", line 374, in train\r\n    scaled_loss.backward()\r\n  File \"venv/lib/python3.7/site-packages/torch/tensor.py\", line 195, in backward\r\n    torch.autograd.backward(self, gradient, retain_graph, create_graph)\r\n  File \"venv/lib/python3.7/site-packages/torch/autograd/__init__.py\", line 99, in backward\r\n    allow_unreachable=True)  # allow_unreachable flag\r\n  File \"venv/lib/python3.7/site-packages/wandb/wandb_torch.py\", line 256, in <lambda>\r\n    handle = var.register_hook(lambda grad: _callback(grad, log_track))\r\n  File \"venv/lib/python3.7/site-packages/wandb/wandb_torch.py\", line 254, in _callback\r\n    self.log_tensor_stats(grad.data, name)\r\n  File \"venv/lib/python3.7/site-packages/wandb/wandb_torch.py\", line 165, in log_tensor_stats\r\n    flat = tensor.view(-1)\r\nRuntimeError: view size is not compatible with input tensor's size and stride (at least one dimension spans across two contiguous subspaces). Use .reshape(...) instead.\r\n```\r\n\r\n", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/287/reactions", "total_count": 1, "+1": 1, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/287/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/197", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/197/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/197/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/197/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/197", "id": 555898811, "node_id": "MDU6SXNzdWU1NTU4OTg4MTE=", "number": 197, "title": "Multiple GPU issue", "user": {"login": "elleobrien", "id": 29716724, "node_id": "MDQ6VXNlcjI5NzE2NzI0", "avatar_url": "https://avatars.githubusercontent.com/u/29716724?v=4", "gravatar_id": "", "url": "https://api.github.com/users/elleobrien", "html_url": "https://github.com/elleobrien", "followers_url": "https://api.github.com/users/elleobrien/followers", "following_url": "https://api.github.com/users/elleobrien/following{/other_user}", "gists_url": "https://api.github.com/users/elleobrien/gists{/gist_id}", "starred_url": "https://api.github.com/users/elleobrien/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/elleobrien/subscriptions", "organizations_url": "https://api.github.com/users/elleobrien/orgs", "repos_url": "https://api.github.com/users/elleobrien/repos", "events_url": "https://api.github.com/users/elleobrien/events{/privacy}", "received_events_url": "https://api.github.com/users/elleobrien/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}, {"id": 1807729876, "node_id": "MDU6TGFiZWwxODA3NzI5ODc2", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/stale", "name": "stale", "color": "fcf8a9", "default": false, "description": "This issue has become stale"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 9, "created_at": "2020-01-27T23:37:55Z", "updated_at": "2021-01-25T18:10:38Z", "closed_at": "2021-01-25T18:10:38Z", "author_association": "NONE", "active_lock_reason": null, "body": "Hello, this is a wonderful library and has been very easy for me to use so far! The only issue I'm having is going from 1 -> multiple GPUs. I am doing binary classification (Torch version = 1.2.0; also tried Torch 1.4.0)\r\n\r\nWhen I set n_gpu = 4, as is the case on my EC2 machine, I get this error:\r\n\r\n```Traceback (most recent call last):                                               | 0/30976 [00:00<?, ?it/s]\r\n  File \"train_classifier.py\", line 46, in <module>\r\n    model.train_model(train_df, eval_df = test_df, show_running_loss=False, acc=accuracy_score)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/simpletransformers/classification/classification_model.py\", line 256, in train_model\r\n    **kwargs,\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/simpletransformers/classification/classification_model.py\", line 346, in train\r\n    outputs = model(**inputs)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/modules/module.py\", line 547, in __call__\r\n    result = self.forward(*input, **kwargs)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py\", line 152, in forward\r\n    outputs = self.parallel_apply(replicas, inputs, kwargs)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py\", line 162, in parallel_apply\r\n    return parallel_apply(replicas, inputs, kwargs, self.device_ids[:len(replicas)])\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/parallel/parallel_apply.py\", line 85, in parallel_apply\r\n    output.reraise()\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/_utils.py\", line 369, in reraise\r\n    raise self.exc_type(msg)\r\nRuntimeError: Caught RuntimeError in replica 1 on device 1.\r\nOriginal Traceback (most recent call last):\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/parallel/parallel_apply.py\", line 60, in _worker\r\n    output = module(*input, **kwargs)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/modules/module.py\", line 547, in __call__\r\n    result = self.forward(*input, **kwargs)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/simpletransformers/classification/transformer_models/roberta_model.py\", line 80, in forward\r\n    loss = loss_fct(logits.view(-1, self.num_labels), labels.view(-1))\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/modules/module.py\", line 547, in __call__\r\n    result = self.forward(*input, **kwargs)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/modules/loss.py\", line 916, in forward\r\n    ignore_index=self.ignore_index, reduction=self.reduction)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/apex/amp/wrap.py\", line 28, in wrapper\r\n    return orig_fn(*new_args, **kwargs)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/functional.py\", line 1995, in cross_entropy\r\n    return nll_loss(log_softmax(input, 1), target, weight, None, ignore_index, None, reduction)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/apex/amp/wrap.py\", line 28, in wrapper\r\n    return orig_fn(*new_args, **kwargs)\r\n  File \"/home/ubuntu/anaconda3/lib/python3.6/site-packages/torch/nn/functional.py\", line 1824, in nll_loss\r\n    ret = torch._C._nn.nll_loss(input, target, weight, _Reduction.get_enum(reduction), ignore_index)\r\nRuntimeError: Assertion `THCTensor_(checkGPU)(state, 5, input, target, weights, output, total_weight)' failed. Some of weight/gradient/input tensors are located on different GPUs. Please move them to a single one. at /pytorch/aten/src/THCUNN/generic/ClassNLLCriterion.cu:24\r\n```\r\n\r\nI have seen a previous issue posted about getting an error with multiple GPUs when the running loss is displayed, but even with `show_running_loss=False`, this error is occurring. My guess is that something is breaking when it comes time to compute the loss? Any ideas?", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/197/reactions", "total_count": 1, "+1": 1, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/197/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/68", "repository_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers", "labels_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/68/labels{/name}", "comments_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/68/comments", "events_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/68/events", "html_url": "https://github.com/ThilinaRajapakse/simpletransformers/issues/68", "id": 530648849, "node_id": "MDU6SXNzdWU1MzA2NDg4NDk=", "number": 68, "title": "Muliti GPU training", "user": {"login": "aadhavan-account", "id": 10369551, "node_id": "MDQ6VXNlcjEwMzY5NTUx", "avatar_url": "https://avatars.githubusercontent.com/u/10369551?v=4", "gravatar_id": "", "url": "https://api.github.com/users/aadhavan-account", "html_url": "https://github.com/aadhavan-account", "followers_url": "https://api.github.com/users/aadhavan-account/followers", "following_url": "https://api.github.com/users/aadhavan-account/following{/other_user}", "gists_url": "https://api.github.com/users/aadhavan-account/gists{/gist_id}", "starred_url": "https://api.github.com/users/aadhavan-account/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/aadhavan-account/subscriptions", "organizations_url": "https://api.github.com/users/aadhavan-account/orgs", "repos_url": "https://api.github.com/users/aadhavan-account/repos", "events_url": "https://api.github.com/users/aadhavan-account/events{/privacy}", "received_events_url": "https://api.github.com/users/aadhavan-account/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 1594703670, "node_id": "MDU6TGFiZWwxNTk0NzAzNjcw", "url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/labels/bug", "name": "bug", "color": "d73a4a", "default": true, "description": "Something isn't working"}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2019-12-01T00:31:32Z", "updated_at": "2019-12-03T17:53:27Z", "closed_at": "2019-12-03T17:53:27Z", "author_association": "NONE", "active_lock_reason": null, "body": "In the file classification_model.py, under the method train in ClassificationModel, you should move the script to display running average below the if statement used to check if there are multiple GPU's used for training. Or in other words you have to average the loss before displaying it.", "reactions": {"url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/68/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/ThilinaRajapakse/simpletransformers/issues/68/timeline", "performed_via_github_app": null, "state_reason": "completed"}]